### ISSUE TITLE
15.0.0-rc2 - upgrade 14 to 15 - SQLite DB table locked

### ISSUE THREAD

**Issue by mjpraxis** [umbraco/Umbraco-CMS#17400]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

15.0.0-rc2

### Bug summary

Updating a v14 database to a v15rc2

The console log indicates that there is some properties being removed - that totally fine I guess - great that the upgrade process can clean up.

But then it looks like that the SQLite connection can not handle multiple connections when making changes to the table umbracoPropertyData.

I am not sure how to reproduce this scenario. I some how got these properties that the upgrade flow flags and wants to clean up.

My guess is this could be that there is a list of threads started to make these DB changes. And that the SQLite is not happy with so many connections. Probably there will be no problems if the database was a "real" MS SQL server that can handle several connections.

Solution?: 
- Combine all the SQL command to one-command and execute that on the SQLite DB
- Do not run in threads when it is a SQLite provider.

### Steps to reproduce

**Console log**
Reduced - attached is the complete log
[consolelog.txt](https://github.com/user-attachments/files/17587785/consolelog.txt)

```
[13:05:09 INF] Checking if DatabaseUpgradeStep requires execution
[13:05:09 INF] Running DatabaseUpgradeStep
[13:05:09 INF] Running 'Upgrade' service
[13:05:09 INF] Database upgrade started
[13:05:09 INF] Starting 'Umbraco.Core'...
[13:05:09 INF] At {6C04B137-0097-4938-8C6A-276DF1A0ECA8}
[13:05:09 INF] Execute ConvertBlockGridEditorProperties
[13:05:09 INF] Migration starting for all properties of type: Umbraco.BlockGrid
[13:05:10 INF] - starting property type: Dekoration (id: 56, alias: decoration)...
[13:05:10 INF]   - 4 properties converted, saving...
[13:05:10 INF] - starting property type: Indhold (id: 55, alias: content)...
[13:05:10 WRN] The property contentPartKey for block 739b315b-5735-4dd8-9958-93f75a72e79e was removed because the property type was not found on webbog_contentLayout_twoColumns
[13:05:10 WRN] The property contentPartKey for block 22cf7de3-3da7-4221-acb8-6163fc6e728e was removed because the property type was not found on webbog_contentLayout_boxhalfwidth
[13:05:10 WRN] The property contentPartKey for block 739b315b-5735-4dd8-9958-93f75a72e79e was removed because the property type was not found on webbog_contentLayout_twoColumns
[13:05:10 WRN] The property contentPartKey for block 0523cd15-f4b8-4014-9148-d1d68df74e1f was removed because the property type was not found on webbog_contentLayout_assignment
[13:05:10 WRN] The property contentPartKey for block 0523cd15-f4b8-4014-9148-d1d68df74e1f was removed because the property type was not found on webbog_contentLayout_assignment
[13:05:10 WRN] The property contentPartKey for block 10599d80-9928-413b-9ffd-805bc444054c was removed because the property type was not found on webbog_contentLayout_boxfullwidth
[13:05:10 WRN] The property contentPartKey for block dbb06de9-e239-49be-9c5c-8fea5ec25c5b was removed because the property type was not found on webbog_contentLayout_boxhalfwidth
[13:05:11 WRN] The property contentPartKey for block 0523cd15-f4b8-4014-9148-d1d68df74e1f was removed because the property type was not found on webbog_contentLayout_assignment
[13:05:11 WRN] The property contentPartKey for block 10599d80-9928-413b-9ffd-805bc444054c was removed because the property type was not found on webbog_contentLayout_boxfullwidth
[13:05:11 WRN] The property contentPartKey for block 22cf7de3-3da7-4221-acb8-6163fc6e728e was removed because the property type was not found on webbog_contentLayout_boxhalfwidth
[13:05:11 WRN] The property contentPartKey for block dbb06de9-e239-49be-9c5c-8fea5ec25c5b was removed because the property type was not found on webbog_contentLayout_boxhalfwidth
[13:05:11 WRN] The property contentPartKey for block 739b315b-5735-4dd8-9958-93f75a72e79e was removed because the property type was not found on webbog_contentLayout_twoColumns
[13:05:11 WRN] The property contentPartKey for block 0523cd15-f4b8-4014-9148-d1d68df74e1f was removed because the property type was not found on webbog_contentLayout_assignment
[13:14:47 ERR] Exception (0824c6a9).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at StackExchange.Profiling.Data.ProfiledDbCommand.ExecuteDbDataReader(CommandBehavior behavior) in C:\projects\dotnet\src\MiniProfiler.Shared\Data\ProfiledDbCommand.cs:line 204
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.RetryPolicy.ExecuteAction[TResult](Func`1 func)
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.Execute[T](Func`1 f)
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at NPoco.Database.<>c__DisplayClass298_0.<ExecuteReaderHelper>b__0()
   at NPoco.Database.ExecutionHook[T](Func`1 action)
   at NPoco.Database.ExecuteReaderHelper(DbCommand cmd)
   at NPoco.Database.ExecuteDataReader(DbCommand cmd, Boolean sync)
   at NPoco.AsyncHelper.RunSync[T](Task`1 task)
   at NPoco.Database.QueryImp[T](T instance, Expression`1 listExpression, Func`2 idFunc, Sql Sql, PocoData pocoData)+MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at NPoco.Database.Fetch[T](Sql sql)
   at Umbraco.Extensions.NPocoDatabaseExtensions.<>c__DisplayClass13_0`2.<FetchByGroups>b__0(IEnumerable`1 x)
   at System.Linq.Enumerable.IEnumerableSelectIterator`2.MoveNext()
   at Umbraco.Extensions.EnumerableExtensions.SelectByGroups[TResult,TSource](IEnumerable`1 source, Func`2 selector, Int32 groupSize)+MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.ContentRepositoryBase`3.GetPropertyCollections[T](List`1 temps)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.MediaRepository.MapDtoToContent(ContentDto dto)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.MediaRepository.MediaByGuidReadRepository.PerformGet(Guid id)
   at Umbraco.Cms.Core.Cache.DefaultRepositoryCachePolicy`2.Get(TId id, Func`2 performGet, Func`2 performGetAll)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.EntityRepositoryBase`2.Get(TId id)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.MediaRepository.Get(Guid id)
   at Umbraco.Cms.Core.Services.MediaService.GetById(Guid key)
   at Umbraco.Cms.Core.PropertyEditors.MediaPicker3PropertyEditor.MediaPicker3PropertyValueEditor.UpdateMediaTypeAliases(List`1 mediaWithCropsDtos)
   at Umbraco.Cms.Core.PropertyEditors.MediaPicker3PropertyEditor.MediaPicker3PropertyValueEditor.ToEditor(IProperty property, String culture, String segment)
   at Umbraco.Cms.Core.PropertyEditors.BlockValuePropertyValueEditorBase`2.MapBlockItemDataToEditor(IProperty property, List`1 items, String culture, String segment)
   at Umbraco.Cms.Core.PropertyEditors.BlockValuePropertyValueEditorBase`2.MapBlockValueToEditor(IProperty property, TValue blockValue, String culture, String segment)
   at Umbraco.Cms.Core.PropertyEditors.BlockEditorPropertyValueEditor`2.ToEditor(IProperty property, String culture, String segment)
   at Umbraco.Cms.Infrastructure.Migrations.Upgrade.V_15_0_0.ConvertBlockEditorPropertiesBase.<>c__DisplayClass18_2.<Handle>b__2(UpdateBatch`1 update)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
[13:14:47 ERR] Exception (ebffb223).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:49 ERR] Exception (250ec4c5).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:50 ERR] Exception (4a9d4eca).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:51 ERR] Exception (c7e2c5da).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:51 ERR] Exception (bef8a7d9).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:52 ERR] Exception (ba0bca74).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:52 ERR] Exception (fa13ceb5).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:54 ERR] Exception (bb6e8332).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:54 ERR] Exception (c340e780).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:54 ERR] Exception (d2c2eba2).
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:54 ERR] Migration failed for property type: Indhold (id: 55, alias: content, editor alias: Umbraco.BlockGrid)
System.AggregateException: One or more errors occurred. (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.) (SQLite Error 6: 'database table is locked: umbracoPropertyData'.)
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 6: 'database table is locked: umbracoPropertyData'.

[13:14:54 ERR] Migration failed for one or more properties of type: Umbraco.BlockGrid
[13:14:54 INF] At {9D3CE7D4-4884-41D4-98E8-302EB6CB0CF6}
[13:14:54 INF] Execute ConvertRichTextEditorProperties
[13:14:54 INF] Migration starting for all properties of type: Umbraco.RichText
[13:14:54 INF] - starting property type: TextHtml (id: 86, alias: textHtml)...
[13:14:54 INF]   - skipped property type migration because it was not a applicable.
[13:14:54 INF] - starting property type: Fodnote (id: 84, alias: footnote)...
[13:14:54 INF]   - skipped property type migration because it was not a applicable.
[13:14:54 INF] Migration succeeded for all properties of type: Umbraco.RichText
[13:14:54 INF] At {37875E80-5CDD-42FF-A21A-7D4E3E23E0ED}
[13:14:54 INF] Done
[13:14:54 INF] Database configuration status: <p>Upgrade completed!</p>
[13:14:54 INF] Finished DatabaseUpgradeStep
```

**Comment by mjpraxis** [comment]
If you need a copy of my v14 SQLite database, I can send it to you.

**Comment by bergmania** [comment]
Sounds like we potentially cannot execute in parallel when using SQLite.

**Comment by kjac** [comment]
Hi @mjpraxis - I'd love to get my hands on your SQLite DB to test a fix 😄 

Can you send it to kja@umbraco.dk?

**Comment by mjpraxis** [comment]
@bergmania @kjac 
Yes, my experience with SQLite and parallel is not good.

This one https://github.com/umbraco/Umbraco-CMS/blob/7faceb365b9cd32a702c2e9b888a5534e6f57b24/src/Umbraco.Infrastructure/Migrations/Upgrade/V_15_0_0/ConvertBlockEditorPropertiesBase.cs#L154
could be the troublemaker

**Comment by mjpraxis** [comment]
> Hi @mjpraxis - I'd love to get my hands on your SQLite DB to test a fix 😄
> 
> Can you send it to [kja@umbraco.dk](mailto:kja@umbraco.dk)?

Database is in your inbox :-)

**Comment by kjac** [comment]
Fix in #17416 👍
